examples/model_selection/plot_grid_search_stats.py

```diff
@@ -0,0 +1,568 @@
+Statistical comparison of models using grid search
+==================================================
+
+This example illustrates how to statistically compare the performance of models
+trained and evaluated using :class:`~sklearn.model_selection.GridSearchCV`.
+
+"""
+
+###############################################################################
+# We will start by simulating moon shaped data (where the ideal separation
+# between classes is non-linear), adding to it a moderate degree of noise.
+# Datapoints will belong to one of two possible classes represented
+# by two features. We will simulate 50 samples for each class:
+
+print(__doc__)
+import matplotlib.pyplot as plt
+import seaborn as sns
+from sklearn.datasets import make_moons
+
+X, y = make_moons(noise=0.352, random_state=1, n_samples=100)
+
+sns.scatterplot(
+    X[:, 0], X[:, 1], hue=y,
+    marker='o', s=25, edgecolor='k', legend=False
+).set_title("Data")
+plt.show()
+
+
+###############################################################################
+# We will compare the performance of SVC estimators that vary on their `kernel`
+# parameter, to decide which choice of this hyper-parameter predicts our
+# simulated data best. We will evaluate the performance of the models using
+# :class:`~sklearn.model_selection.RepeatedStratifiedKFold`, repeating 10 times
+# a 10-fold stratified cross validation using a different randomization of the
+# data in each repetition. The performance will be evaluated using
+# :class:`~sklearn.metrics.roc_auc_score`.
+
+from sklearn.model_selection import GridSearchCV, RepeatedStratifiedKFold
+from sklearn.svm import SVC
+
+param_grid = [
+    {'kernel': ['linear']},
+    {'kernel': ['poly'], 'degree': [2, 3]},
+    {'kernel': ['rbf']}
+]
+
+svc = SVC(random_state=0)
+
+cv = RepeatedStratifiedKFold(
+    n_splits=10, n_repeats=10, random_state=0
+)
+
+search = GridSearchCV(
+    estimator=svc, param_grid=param_grid,
+    scoring='roc_auc', cv=cv
+)
+search.fit(X, y)
+
+
+###############################################################################
+# We can now inspect the results of our search, sorted by their
+# `mean_test_score`:
+
+import pandas as pd
+
+results_df = pd.DataFrame(search.cv_results_)
+results_df = (
+    results_df
+    .sort_values(by=['rank_test_score'])
+    .set_index(results_df["params"].apply(
+        lambda x: "_".join(str(val) for val in x.values()))
+    )
+    .rename_axis('model')
+)
+results_df[
+    ['params', 'rank_test_score', 'mean_test_score', 'std_test_score']
+]
+
+
+###############################################################################
+# We can see that the estimator using the `rbf` kernel performed better,
+# closely followed by `linear`. Both estimators with a `poly` kernel performed
+# worse, with the one using a two degree polynomial achieving a much lower
+# perfomance than all other models.
+#
+# The output of `GridSearchCV` does not provide information on the certainty of
```

Comment:
This is the quintessence of all that follows! I miss a sentence in boldface. Something like: "This is all nice, but we don't know if the differences are statistically significant at all or just random."